src/core/qgsdistancearea.cpp

```diff
@@ -57,6 +57,9 @@ bool QgsDistanceArea::willUseEllipsoid() const
 
 void QgsDistanceArea::setSourceCrs( const QgsCoordinateReferenceSystem &srcCRS )
 {
+  mSemiMajor = -1.0;
```

Comment:
Why is this here? It means settings the sourceCrs after setting an ellipsoid will lose the ellipsoid settings